### PR TITLE
Address issue #2083, fixing a time-zone bug in a unit test

### PR DESCRIPTION
**Pull request by mallman**
Address issue #2083, fixing a time-zone related bug in test_issue_1959.py.

Note, I'm following the documentation in the [exiv2.md](https://github.com/Exiv2/exiv2/blob/46c329081f147d68ba38e7256481d8e432cac64c/exiv2.md#TZ) in setting the TZ variable.

### CODE REVIEW

**Comment by mallman** [comment]
Derp. Guess this patch isn't going to work so great on Windows. I'll work on a more robust approach.

**Comment by hassec** [comment]
@piponazo I think this conflicts with your #2079   

@mallman thanks a lot for filing the issue and directly working on it :+1:    
I do like where your solution is going, but wonder if we can somehow fix the timezone globally for the entire testsuite.

**Comment by piponazo** [comment]
Hehe it is funny that we ended up trying to fix the same issue on the same day. Another wonder of the FOSS community 😇 .

I do not believe this is the first case where such problem arised. @clanmills Maybe you can guide us on this topic. Did you ever encountered such problems? (The one described at #2083)

File: tests/bugfixes/github/test_issue_1959.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def setUp(self):
     filename      = path("$data_path/issue_1959_poc.xmp")
     filename_ref  = path("$data_path/issue_1959_poc.xmp.out")
     
-    commands = ["$exiv2 -Pkvt $filename"]
+    commands = ["TZ=UTC $exiv2 -Pkvt $filename"]
```
Comment (hassec) [inline tests/bugfixes/github/test_issue_1959.py:18]: @piponazo @mallman I seem to have found the existing setting we need to use for this :tada: 


```suggestion
    env = {
        'TZ': 'UTC'
    }
    commands = ["$exiv2 -Pkvt $filename"]
```

Comment (piponazo) [reply tests/bugfixes/github/test_issue_1959.py:18]: @mallman if you apply that change and it works, I'll close #2079 .

Comment (mallman) [reply tests/bugfixes/github/test_issue_1959.py:18]: WFM. Let's see if the Windows tests pass...

**Comment by mallman** [comment]
Actually, I'm putting this on hold while I investigate a closely related matter.

**Comment by mallman** [comment]
I've put this PR on hold, because I'm not sure this test is well-defined. Why is the system's timezone (or value of TZ) relevant in determining Exif.Photo.DateTimeOriginal in this test case?

According to the documentation of Exif.Photo.DateTimeOriginal on https://exiv2.org/tags.html, Exif.Photo.DateTimeOriginal is "The date and time when the original image data was generated." To emphasize, it is not _a_ date and time, it is _the_ date and time.

If I shot a photo at 6 AM on 1 January, 2000 in Iceland, shouldn't Exif.Photo.DateTimeOriginal be "2000:01:01 06:00:00"? And if I shot a photo at 6 AM on 1 January, 2000 in Namibia, shouldn't Exif.Photo.DateTimeOriginal also be "2000:01:01 06:00:00"?

Why would my system timezone matter? In both cases, it is the _local_ time zone that determines the value of Exif.Photo.DateTimeOriginal.

And in this particular test case, the only creation date we have is "2022-01-04T09:41:01+00:00". How do we compute the value of Exif.Photo.DateTimeOriginal given that? In examining the source document for this test, I could find no information on the local time zone.

**Comment by clanmills** [comment]
@mallman The TZ environment string should not be involved with Exif.Photo.DateTimeOriginal.  It's an ascii string in the metadata.  The Exif Specification added tags for TZ and milliseconds and these tags are supported in Exiv2.

```
653 rmills@rmillsm1:~/gnu/github/exiv2/main/build $ taglist all | grep Original | grep Photo
Photo.DateTimeOriginal
Photo.OffsetTimeOriginal
Photo.SubSecTimeOriginal
654 rmills@rmillsm1:~/gnu/github/exiv2/main/build $
```

Exiv2 should not post-process that metadata in any way.  It don't see any evidence that it does.  I will inspect the code. 

```
656 rmills@rmillsm1:~/gnu/github/exiv2/main/build $ env TZ=GMT exiv2 -Pv -K Exif.Photo.DateTimeOriginal ~/Stonehenge.jpg 
2015:07:16 15:38:54
657 rmills@rmillsm1:~/gnu/github/exiv2/main/build $ env TZ=PST8PDT exiv2 -Pv -K Exif.Photo.DateTimeOriginal ~/Stonehenge.jpg 
2015:07:16 15:38:54
658 rmills@rmillsm1:~/gnu/github/exiv2/main/build $ 
```

Thank You for opening #2082.  I've discussed this with @piponazo and @kmilos and will review/update README.md and CONTRIBUTING.md over the weekend.  I haven't heard if @hassec did service in the Swiss Army.

**Comment by clanmills** [comment]
@mallman Apologies.  I hadn't looked at the test when I wrote about Exif.Photo.DateTimeOriginal.

I now understand that this is an XMP test.  There is date information in that file:

```bash
713 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ xmllint --pretty 2 issue_1959_poc.xmp | grep -i date
        photoshop:DateCreated="2022-01-04T09:41:01+00:00"
                Iptc4xmpExt:AOCircaDateCreated="2022-01-04"
                Iptc4xmpExt:AODateCreated="2022-01-04T09:41:01+00:00"
714 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ 
```
That's being "converted" into Exif.Photo.DateTimeOriginal:

```
713 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ xmllint --pretty 2 issue_1959_poc.xmp | grep -i date
        photoshop:DateCreated="2022-01-04T09:41:01+00:00"
                Iptc4xmpExt:AOCircaDateCreated="2022-01-04"
                Iptc4xmpExt:AODateCreated="2022-01-04T09:41:01+00:00"
714 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ exiv2 -pe issue_1959_poc.xmp
Exif.Image.YCbCrPositioning                  Short       1  1
Exif.Image.XResolution                       Rational    1  72/1
Exif.Image.YResolution                       Rational    1  72/1
Exif.Image.ResolutionUnit                    Short       1  2
Exif.Image.ImageDescription                  Ascii      32  Test file for the IPTC XMP tags
Exif.Image.Artist                            Ascii      15  postscript-dev
Exif.Image.Copyright                         Ascii      16  Copyright Exiv2
Exif.Photo.ExifVersion                       Undefined   4  48 50 51 50
Exif.Photo.FlashpixVersion                   Undefined   4  48 49 48 48
Exif.Photo.ColorSpace                        Short       1  65535
Exif.Photo.ComponentsConfiguration           Undefined   1  1
Exif.Photo.DateTimeOriginal                  Ascii      20  2022:01:04 09:41:01
715 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ 
```

In the dungeons of Exiv2, there are metadata "convertors" what attempt to synchronise data between different standards (Exif, Iptc and Xmp).  I don't know who thought of this bad idea.  

I've just change the timezone on my MacBook Pro to the West Coast (I lived in San Jose for 15 years.  And shock/horror:

```bash
715 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ exiv2 -pe issue_1959_poc.xmp > ~/temp/England.txt
... use SysPrefs to move to California ...
717 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ date
Thu 10 Feb 2022 11:02:58 PST
718 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ exiv2 -pe issue_1959_poc.xmp > ~/temp/California.txt
719 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ diff ~/temp/England.txt ~/temp/California.txt 
12c12
< Exif.Photo.DateTimeOriginal                  Ascii      20  2022:01:04 09:41:01
---
> Exif.Photo.DateTimeOriginal                  Ascii      20  2022:01:04 01:41:01
720 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ 
```
A lunatic is using a TZ time conversion function.  That's obviously wrong.  Wrong, wrong, wrong.  Shsshssssss.

I don't need the output I request before.  I can easily generate it:

```bash
724 rmills@rmillsm1:~/gnu/github/exiv2/main/tests $ python3 runner.py bugfixes/github/*1959*.py 2>&1 | wc -l
     250
725 rmills@rmillsm1:~/gnu/github/exiv2/main/tests $ python3 runner.py bugfixes/github/*1959*.py 
...
  Xmp.photoshop.CaptionWriter                   Test Name  Test Name
  Xmp.dc.title                                  lang="x-default" Test IPTC XMP file  lang="x-default" Test IPTC XMP file
  Xmp.dc.subject                                Test  Test
  Xmp.dc.creator                                postscript-dev  postscript-dev
  Xmp.dc.rights                                 lang="x-default" Copyright Exiv2  lang="x-default" Copyright Exiv2
  Xmp.dc.description                            lang="x-default" Test file for the IPTC XMP tags  lang="x-default" Test file for the IPTC XMP tags
 : Standard output does not match

----------------------------------------------------------------------
Ran 1 test in 0.046s

FAILED (failures=1)
       0       0       0
724 rmills@rmillsm1:~/gnu/github/exiv2/main/tests $
```

**Comment by mallman** [comment]
@clanmills Would you agree with me that the output of exiv2 in this case should not include the Exif.Photo.DateTimeOriginal tag?

**Comment by clanmills** [comment]
Again, I've only just discovered your observations.  More work ahead for Team Exiv2.

**Comment by mallman** [comment]
I've raised my concerns around the inference of xmp timestamp data into `DateTimeOriginal`. (It's somewhat ironic that timestamp conversion even arose in this issue because the intent of the unit test in question is not to test timestamp conversion.) It's been a week since I opened this PR and we seem to have a working solution. In the interests of keeping on point, I'm going to take this PR out of draft and request a merge from a reviewer.

Questions around timestamp/local time conversion can be raised and discussed in a separate, focused issue another time. Cheers!

**Comment by mallman** [comment]
`test_issue_1959.py` was deleted in b318b9d, rendering this PR moot. Closing without merge.

**Comment by clanmills** [comment]
@mallman.  Thanks for this.  Gosh, you're a clever guy to have spotted out.  Thanks.

I think I will restore test_issue_1959.py and submit a PR in which I remove the `photoshop` entry in the side-car.  However, I'll do this when I understand why test_issue_1959.py was deleted.

Thanks also for your PR about Nikon/AF2.  What you've done seems right to me, however I don't want to get sucked into that issue.

**Comment by piponazo** [comment]
> I think I will restore test_issue_1959.py and submit a PR in which I remove the `photoshop` entry in the side-car. However, I'll do this when I understand why test_issue_1959.py was deleted.

Hi Robin. That test was removed in #2091 (you can see the discussion there)

**Comment by clanmills** [comment]
@mallman I could really enjoy working with you.  Yes.  I think removing the 1959 tests in #2091 has something to do with babies and bathwater _(or in my case drowning the grandkids in the swimming pool)_.  I'm going to submit a PR which I think will be accepted without using more of your time.  Thanks very much.

**Comment by clanmills** [comment]
#2103

**Pull request closed without merging**